iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -220,6 +220,20 @@ public override int GetHashCode()
             return hashCode;
         }
 
+        // Edge Modules and Module Twins have different links to be used for the same function when communicating over AMQP
+        // We are setting the flag on these methods since the decision should be made at the transport layer and not at the
+        // client layer.
+        //
+        // This means that all other transports will need to implement this method. However they do not need to use the flag
+        // if there is no behavior change required.
+        internal bool IsEdgeModule
```

Comment:
This logic used to sit in module client, but having a getter here simplifies things for our MQTT and AMQP layers who need to know this flag